### PR TITLE
Use Forward Declaration instead of direct imports

### PR DESCRIPTION
**Pull request by AlexDenisov**
[Here is](http://railsware.com/blog/2013/08/09/using-forward-declaration-in-your-objective-c-projects/) you can find explanation about pros of forward declaration and cons of imports in headers.

### CODE REVIEW

File: Classes/CPConfig.h [hunk 1]
```diff
@@ -7,7 +7,9 @@
 //
 
 #import <Foundation/Foundation.h>
-#import "CPSourceAggreagate.h"
+
+@class CPSourceAggreagate;
```
Comment (orta) [inline Classes/CPConfig.h:11]: `CPSourceAggreagate` -> `CPSourceAggregate`

**Comment by orta** [comment]
Why did you remove the Podfile.lock?

File: Classes/CPConfig.h (hunk 1 above)
Comment (AlexDenisov) [inline Classes/CPConfig.h:11]: There is a typo in the class name., see `import` above.

**Comment by AlexDenisov** [comment]
I thought that this file should not be stored in a repo. 
I can revert this change and put it back. Anyway, could you, please, explain why this file should be stored in repo?

**Comment by orta** [comment]
The Podfile.lock is the promise that you have the same version as everyone else. If you use ambiguous dependencies like `pod "ARAnalytics", "~>1.3"` that could be version 1.3 to 1.9. The Podfile.lock says which version was used last time.

File: Classes/CPConfig.h (hunk 1 above)
Comment (orta) [inline Classes/CPConfig.h:11]: Noted, I've made an issue for it.

**Comment by orta** [comment]
If you revert the Podfile.lock change, I'm cool to merge. I'm also a big fan of forward declarations.

**Comment by AlexDenisov** [comment]
Thank you for explanation.
`Podfile.lock` reverted back.

**Comment by fabiopelosin** [comment]
:+1:

**Pull request merged**